src/lib/components/Creatibutors/CreatibutorsModal.js

```diff
@@ -171,6 +171,8 @@ export class CreatibutorsModal extends Component {
     formikBag.resetForm();
     switch (this.state.action) {
       case 'saveAndContinue':
+        this.closeModal();
```

Comment:
Can we do instead:
```
formikBag.resetForm();
this.focusInput();
```
?

Reply:
This isn't working unfortunately mainly because the problem is not in the `formik` values, the issue is with the [TextFields](https://github.com/inveniosoftware/react-invenio-forms/blob/master/src/lib/TextField.js) because it manages the state internally and only updates the formik values [onBlur](https://github.com/inveniosoftware/react-invenio-forms/blob/master/src/lib/TextField.js#L41-L47). If we keep the modal open and only reset the formik values the internal state of `TextField` won't reset and keep the previous values. Maybe another way to do it is to check if formik values were submitted evey time we update the [TextField](https://github.com/inveniosoftware/react-invenio-forms/blob/master/src/lib/TextField.js) and if so, reset the state, WDYT?

Reply:
Let's keep it like this for the time being. Modals should be refactored to avoid this too and handle the state correctly.

I would simply add a comment to explain why this is needed because otherwise it looks weird, push and merge :)